cli/src/parse.rs

```diff
@@ -277,6 +337,21 @@ fn parse_edit_flag(source_code: &Vec<u8>, flag: &str) -> Result<Edit> {
     })
 }
 
+fn offset_for_lines(input: &Vec<u8>) -> Vec<usize> {
+    let mut offsets = Vec::<usize>::new();
+    offsets.push(0);
+    for (i, c) in input.iter().enumerate() {
+        if *c as char == '\n' {
+            offsets.push(i + 1);
+        }
+    }
+    offsets
+}
```

Comment:
If I understand correctly, you wouldn't need these functions if you updated your code in `parse_file_at_path` to just use the `Node::start_byte` and `Node::end_byte` methods.

Reply:
These are really great suggestions, Max, making my changes much simplified. I have just fixed those and reverted the changes to the original code introduced by `cargo-fm`.